standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1792,6 +1807,9 @@ private ValidTxnList getValidTxnList(Connection dbConn, String fullTableName, Lo
     PreparedStatement pst = null;
     ResultSet rs = null;
     try {
+      // To reuse dbConn, we should get open transactions at first
+      GetOpenTxnsResponse openTxns = getOpenTxns(dbConn);
```

Comment:
we should have lazy/conditional evaluation here based on the select statement, just move it under the if block
```
TxnCommonUtils.createValidReadTxnList(getOpenTxns(dbConn), rs.getLong(1))
```